Makefile

```diff
@@ -93,82 +216,93 @@ SONG_OBJS := $(patsubst $(SONG_SUBDIR)/%.s,$(SONG_BUILDDIR)/%.o,$(SONG_SRCS))
 endif
 
-tools: $(TOOLDIRS)
-
-$(TOOLDIRS):
-	@$(MAKE) -C $@
-
-# For contributors to make sure a change didn't affect the contents of the ROM.
-compare: all
+syms: $(SYM)
 
 SUBGAME_LOADERS := multi_boot/subgame_loaders
 
-mostlyclean: tidy
-	@rm -f sound/direct_sound_samples/*.bin
-	@rm -f $(SONG_OBJS) $(MID_SUBDIR)/*.s
-	@find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
-	@rm -f $(AUTO_GEN_TARGETS)
+clean: tidy clean-tools clean-generated clean-assets
 	@$(MAKE) -C $(SUBGAME_LOADERS) $@
 	@$(MAKE) -C multi_boot/unk_8D94B9C $@
 	@$(MAKE) -C multi_boot/unk_8E1FE28 $@
 	@$(MAKE) -C multi_boot/unk_8E8490C $@
 
-clean-tools:
-	@$(foreach tooldir,$(TOOLDIRS),$(MAKE) clean -C $(tooldir);)
-
-clean: mostlyclean clean-tools
+clean-assets:
+	@echo "Cleaning $(FILE_NAME) assets"
+	@echo "rm -f $(MID_SUBDIR)/*.s"
+	@rm -f $(MID_SUBDIR)/*.s
+	@echo "rm -f sound/direct_sound_samples/*.bin"
+	@rm -f sound/direct_sound_samples/*.bin
+	@echo "find -iname '<graphics-related>' -exec rm {} +"
+	@find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
 	@$(MAKE) -C $(SUBGAME_LOADERS) $@
 	@$(MAKE) -C multi_boot/unk_8D94B9C $@
 	@$(MAKE) -C multi_boot/unk_8E1FE28 $@
 	@$(MAKE) -C multi_boot/unk_8E8490C $@
 
-tidy:
-	@rm -f $(ROM) $(ELF) $(MAP)
-	@rm -r build/*
+tidy: tidynonmodern tidymodern
 	@$(MAKE) -C $(SUBGAME_LOADERS) $@
 	@$(MAKE) -C multi_boot/unk_8D94B9C $@
 	@$(MAKE) -C multi_boot/unk_8E1FE28 $@
 	@$(MAKE) -C multi_boot/unk_8E8490C $@
 
+tidynonmodern:
+	@rm -f $(ROM_NAME) $(ELF_NAME) $(MAP_NAME)
+	@rm -rf $(OBJ_DIR_NAME)
+
+tidymodern:
+	@rm -f $(MODERN_ROM_NAME) $(MODERN_ELF_NAME) $(MODERN_MAP_NAME)
+	@rm -rf $(MODERN_OBJ_DIR_NAME)
+
+# Other rules
 include graphics_file_rules.mk
-include songs.mk
 
-%.s: ;
+# NOTE: Tools must have been built prior (FIXME)
+# so you can't really call this rule directly
+generated: $(AUTO_GEN_TARGETS)
+	@: # Silence the "Nothing to be done for `generated'" message, which some people were confusing for an error.
+
+
+%.s:   ;
 %.png: ;
 %.pal: ;
 %.aif: ;
 
-%.1bpp: %.png  ; $(GFX) $< $@
-%.4bpp: %.png  ; $(GFX) $< $@
-%.8bpp: %.png  ; $(GFX) $< $@
-%.gbapal: %.pal ; $(GFX) $< $@
-%.gbapal: %.png ; $(GFX) $< $@
-%.lz: % ; $(GFX) $< $@
-%.rl: % ; $(GFX) $< $@
+%.1bpp:   %.png  ; $(GFX) $< $@
+%.4bpp:   %.png  ; $(GFX) $< $@
+%.8bpp:   %.png  ; $(GFX) $< $@
+%.gbapal: %.pal  ; $(GFX) $< $@
+%.gbapal: %.png  ; $(GFX) $< $@
+%.lz:     %      ; $(GFX) $< $@
+%.rl:     %      ; $(GFX) $< $@
+
 sound/%.bin: sound/%.aif ; $(AIF) $< $@
 sound/songs/%.s: sound/songs/%.mid
 	cd $(@D) && ../../$(MID) $(<F)
 
+clean-generated:
+	@rm -f $(AUTO_GEN_TARGETS)
+	@echo "rm -f <AUTO_GEN_TARGETS>"
+
 # force targets
 speed_eaters: ;
```

Comment:
These should be phony targets (TBH I don't remember why I created these targets...)

Reply:
Done!